llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -745,8 +761,8 @@ class VSETVLIInfo {
     if (Other.isUnknown())
       return isUnknown();
 
-    if (!hasSameAVL(Other))
-      return false;
+    if (!hasSameAVLLatticeValue(Other))
```

Comment:
There are still some users of == that I think we need to switch over from the notion of lattice value equality to runtime value equality, see VSETVLIInfo:: intersect/needVSETVLIPHI/emitVSETVLIs

They currently work in main because no unknowns reach them but I'm not sure if this will be correct after the nullable ValNo change.

So I think we still need #94340 or something to that effect.